components/Dialogs/src/MDCAlertController.m

```diff
@@ -186,6 +189,48 @@ - (void)addAction:(MDCAlertAction *)action {
   [self.view setNeedsLayout];
 }
 
+- (BOOL)mdc_adjustsFontForContentSizeCategory {
+  return _mdc_adjustsFontForContentSizeCategory;
+}
+
+- (void)mdc_setAdjustsFontForContentSizeCategory:(BOOL)adjusts {
+  _mdc_adjustsFontForContentSizeCategory = adjusts;
+
+  // Update any action buttons to match the alert controller.
+  for (MDCButton *actionButton in self.actionButtons) {
+    actionButton.mdc_adjustsFontForContentSizeCategory = adjusts;
+  }
+
+  if (_mdc_adjustsFontForContentSizeCategory) {
+    [self updateFontsForDynamicType];
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(contentSizeCategoryDidChange:)
+                                                 name:UIContentSizeCategoryDidChangeNotification
+                                               object:nil];
+  } else {
+    [[NSNotificationCenter defaultCenter] removeObserver:self
+                                                    name:UIContentSizeCategoryDidChangeNotification
+                                                  object:nil];
+  }
+}
+
+// Handles UIContentSizeCategoryDidChangeNotifications
+- (void)contentSizeCategoryDidChange:(NSNotification *)notification {
+  [self updateFontsForDynamicType];
+}
+
+// Update the fonts used based on mdc_preferredFontForMaterialTextStyle and recalculate the
+// preferred content size.
+- (void)updateFontsForDynamicType {
+  self.titleLabel.font = [UIFont mdc_preferredFontForMaterialTextStyle:MDCFontTextStyleTitle];
```

Comment:
A big motivation for MDCAlertController was to mimic UIAlertController.  Unfortunately this includes it's hostility to subclassing "MDCAlertController class is intended to be used as-is and does not support subclassing. The view hierarchy for this class is private and must not be modified."
We may be able to expose this functionality via the UIAppearance proxy.
I've opened an issue #1215 to investigate a solution.